File: csharp/src/Common/Helpers/RedisHelper.cs
namespace Train.Solver.Common.Helpers;

public static class RedisHelper
{
    public const string Delimiter = ":";

    public static string BuildLockKey(string network, string address) 
        => $"{network}{Delimiter}{address}";

    public static string BuildNonceKey(string network, string address) 
        => $"{network}{Delimiter}{address}{Delimiter}currentNonce";

    public static string BuildNodeScoreKey(string network)
        => $"{network}{Delimiter}nodeScoreboard";
}

File: js/src/Blockchain/Blockchain.Abstraction/Infrastructure/RedisHelper/RedisHelper.ts
export const Delimiter = ":";

export function buildLockKey(network: string, address: string, asset?: string): string {
    return asset
        ? `${network}${Delimiter}${address}${Delimiter}${asset}`
        : `${network}${Delimiter}${address}`;
}

export function buildCurrentNonceKey(network: string, address: string, asset?: string): string {
    return asset
        ? `${network}${Delimiter}${address}${Delimiter}${asset}${Delimiter}currentNonce`
        : `${network}${Delimiter}${address}${Delimiter}currentNonce`;
}

export function buildNextNonceKey(network: string, address: string, asset?: string): string {
    return asset
        ? `${network}${Delimiter}${address}${Delimiter}${asset}${Delimiter}nextNonce`
        : `${network}${Delimiter}${address}${Delimiter}nextNonce`;
}